packages/amplify-cli/src/extensions/amplify-helpers/service-select-prompt.js

```diff
@@ -65,6 +65,45 @@ function serviceSelectionPrompt(context, category, supportedServices) {
   return serviceQuestionWalkthrough(context, supportedServices, category);
 }
 
+function datasourceSelectionPrompt(context, category, supportedDatasources) {
```

Comment:
This function is specific to the api category and this code should live inside the category plugin and not the core CLI package.

Reply:
done - moved to within api plugin package.